.github/workflows/connect-npm-deploy.yml

```diff
@@ -0,0 +1,66 @@
+name: "[Deploy] connect npm deploy"
+on:
+  workflow_call:
+    inputs:
+      package:
+        description: Name of the package to deploy.
+        type: choice
+        options:
+          - blockchain-link-types
+          - blockchain-link-utils
+          - blockchain-link
+          - connect-common
+          - transport
+          - utils
+          - utxo-lib
+          - connect-plugin-stellar
+          - connect-plugin-ethereum
+          - analytics
+          - connect-analytics
+          - type-utils
+          - env-utils
+          - protocol
+          - protobuf
+          - schema-utils`
+
+jobs:
+  deploy-npm-beta:
+    name: node
+    runs-on: ubuntu-latest
+    # Only run on branches like: `npm-release/**`
+    # TODO: we probably should add here some environment
+    if: startsWith(github.ref, 'refs/heads/npm-release/')
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: recursive
+
+      - name: Extract branch name
+        run: echo "branch=${GITHUB_HEAD_REF:-${GITHUB_REF#refs/heads/}}" >> $GITHUB_OUTPUT
+        id: extract_branch
+
+      - name: Setup node
+        uses: actions/setup-node@v4
+        with:
+          node-version-file: ".nvmrc"
+          cache: yarn
+
+      - name: Install dependencies
+        run: yarn install --immutable
+
+      - name: Check version
+        run: node ./ci/scripts/check-version ${{ inputs.package }} ${{ steps.extract_branch.outputs.branch }} beta
+
+      - name: Set NPM token
+        env:
+          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
```

Comment:
I just copy this env name from GitLab, they will probably be named differently here.

Reply:
Yes we need to add NPM_TOKEN as a github environment secret for this.